rt/rs/microprofile-client/src/main/java/org/apache/cxf/microprofile/client/MicroProfileClientFactoryBean.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.cxf.jaxrs.client.JAXRSClientFactoryBean;
+import org.apache.cxf.jaxrs.model.ClassResourceInfo;
+import org.apache.cxf.jaxrs.model.FilterProviderInfo;
+import org.apache.cxf.jaxrs.model.ProviderInfo;
+import org.apache.cxf.microprofile.client.proxy.MicroProfileClientProxyImpl;
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+
+public class MicroProfileClientFactoryBean extends JAXRSClientFactoryBean {
+    private final Comparator<ProviderInfo<?>> comparator;
+    private final List<Object> registeredProviders;
+    private Configuration configuration;
+    private ClassLoader proxyLoader;
+    private boolean inheritHeaders;
+
+    public MicroProfileClientFactoryBean(MicroProfileClientConfigurableImpl<RestClientBuilder> configuration,
+                                         String baseUri, Class<?> aClass) {
+        super();
+        this.configuration = configuration.getConfiguration();
+        this.comparator = MicroProfileClientProviderFactory.createComparator(this);
+        super.setAddress(baseUri);
+        super.setServiceClass(aClass);
+        super.setProviderComparator(comparator);
+        registeredProviders = new ArrayList<>();
+        registeredProviders.addAll(processProviders());
+        if (!configuration.isDefaultExceptionMapperDisabled()) {
+            registeredProviders.add(new DefaultResponseExceptionMapper());
+        }
+        registeredProviders.add(new JsonPProvider());
+        super.setProviders(registeredProviders);
+    }
+
+    @Override
+    public void setClassLoader(ClassLoader loader) {
+        super.setClassLoader(loader);
+        this.proxyLoader = loader;
+    }
+
+    @Override
+    public void setInheritHeaders(boolean inheritHeaders) {
+        super.setInheritHeaders(inheritHeaders);
+        this.inheritHeaders = inheritHeaders;
+    }
+
+    @Override
+    protected void initClient(AbstractClient client, Endpoint ep, boolean addHeaders) {
+        super.initClient(client, ep, addHeaders);
+        MicroProfileClientProviderFactory factory = MicroProfileClientProviderFactory.createInstance(getBus(),
+                comparator);
+        factory.setUserProviders(registeredProviders);
+        ep.put(MicroProfileClientProviderFactory.CLIENT_FACTORY_NAME, factory);
+    }
+
+    @Override
+    protected ClientProxyImpl createClientProxy(ClassResourceInfo cri, boolean isRoot,
+                                                ClientState actualState, Object[] varValues) {
+        if (actualState == null) {
+            return new MicroProfileClientProxyImpl(URI.create(getAddress()), proxyLoader, cri, isRoot,
+                    inheritHeaders, varValues);
+        } else {
+            return new MicroProfileClientProxyImpl(actualState, proxyLoader, cri, isRoot,
+                    inheritHeaders, varValues);
+        }
+    }
+
+    Configuration getConfiguration() {
+        return configuration;
+    }
+
+    private Set<Object> processProviders() {
```

Comment:
I suspect this will be useful to move into the main client, since it handles converting filters into `FilterProviderInfo` objects, rather than regular `ProviderInfo`s